sdk/monitor/Azure.Monitor.Query/src/Models/LogQueryResponse.cs

```diff
@@ -1,9 +1,24 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+using Azure.Core;
+
 namespace Azure.Monitor.Query.Models
 {
     internal partial class LogQueryResponse
     {
+        /// <summary> Contains the tables, columns &amp; rows resulting from the query or the error details if the query failed. </summary>
+        [CodeGenMember("Body")]
+        private LogsBatchQueryResult _body;
+
+        public LogsBatchQueryResult Body
+        {
+            get
+            {
+                _body.HasFailed = Status >= 400;
```

Comment:
Would the value of `HasFailed` ever be false when the service operation succeeded? If so, why wouldn't we throw `RequestFailedException` rather than making someone interrogate this property? Is this just used for batches?

Reply:
It's for the batch response were some queries can succeed and some fail. For the single query scenario we always throw on failure.